pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -216,15 +273,32 @@ public boolean registerTableConfigChangeListener(TableConfigChangeListener table
   }
 
   /**
-   * Returns the schema for the given table, or {@code null} if it does not exist.
+   * Returns the schema for the given logical or physical table, or {@code null} if it does not exist.
    */
   @Nullable
   @Override
   public Schema getSchema(String rawTableName) {
+    if (isLogicalTable(rawTableName)) {
```

Comment:
To save overhead for more common case, can we first lookup `_schemaInfoMap` with `rawTableName`, then fallback to logical table indirect schema name